packages/python/plotly/plotly/express/_core.py

```diff
@@ -1988,6 +1988,10 @@ def make_figure(args, constructor, trace_patch=None, layout_patch=None):
     layout_patch = layout_patch or {}
     apply_default_cascade(args)
 
+    # Ignore facet rows and columns when data frame is empty so as to prevent nrows/ncols equaling 0
```

Comment:
This seems like a reasonable fix, but probably belongs near the bottom of the `infer_config` function, just so it's in roughly the same place as most of the other similar mutations of `args`.

While we're on this topic of empty data frames though, are there any other args that fail when the df length is 0? would be good to understand why these specifically fail and if it impacts other categorical-type attributes like `symbol` or somethinng. Maybe there could be a more generic fix?

Reply:
PS: thanks for the PR! :)

Reply:
Ah yes, that makes sense. And you're very welcome. :)

I've moved it into `infer_config` for now, but I'll look into whether there could be a more generic solution.

Reply:
So it looks like `facet_col` and `facet_row` both fail with an empty dataset because they are used in the computation of `ncols` and `nrows`:

https://github.com/plotly/plotly.py/blob/2b6ec1e16bebe260dfc673db869805f7ceb86e00/packages/python/plotly/plotly/express/_core.py#L2014-L2021

Neither `ncols` nor `nrows` can equal 0, hence the failure. As far as I can tell, nothing else in `args` effects either of the two in any other way.

To me, this seems like a fairly specific failure requiring a specific fix, but I'm still far from familiar with the codebase, so I could easily be wrong. :)